woodwork/type_sys/utils.py

```diff
@@ -16,30 +14,24 @@ def col_is_datetime(col, datetime_format=None):
     if _is_spark_series(col):
         col = col.to_pandas()
 
-    if col.dtype.name.find("datetime") > -1 or (
-        len(col) and isinstance(col.head(1), datetime)
-    ):
+    if pd.api.types.is_datetime64_any_dtype(col):
         return True
 
-    # if it can be cast to numeric, it's not a datetime
+    col = col.dropna()
+    if len(col) == 0:
+        return False
+
     try:
-        pd.to_numeric(col, errors="raise")
-    except (ValueError, TypeError):
-        # finally, try to cast to datetime
-        if col.dtype.name.find("str") > -1 or col.dtype.name.find("object") > -1:
-            try:
-                pd.to_datetime(
-                    col,
-                    errors="raise",
-                    format=datetime_format,
-                    infer_datetime_format=True,
-                )
-            except Exception:
-                return False
-            else:
-                return True
+        pd.to_datetime(
+            col.astype(str),
+            errors="raise",
+            format=datetime_format,
+            infer_datetime_format=True,
+        )
+        return True
 
-    return False
+    except Exception:
```

Comment:
I see these are the two exceptions that can be raised: 
<img width="659" alt="image" src="https://user-images.githubusercontent.com/22552445/161830544-caee1624-a0f7-4078-bc0d-f6349ab1fd06.png">

Are we planning on treating both of these equally and returning False for both scenarios? If so, I think it'd be great to add a small comment on that here for future visibility.

Reply:
I think in general it's better to not have generic exception handling `except Exception` and we should be specific about the exceptions that we are handling here

Reply:
Sounds good, updated!

Reply:
Other specific exceptions in woodwork's use case include overflow and type errors